### DISCUSSION TITLE
Relational data and data aggregation

### DISCUSSION THREAD

**Discussion by ansarizafar** [appy-one/acebase-server#45] [General]
Acebase is an amazing project. I have few questions.

1. How can we deal with relational data?
2. How can we aggregate data?
3. Is it possible to use Acebase server with Bun https://bun.sh/

**Comment by appy-one** [comment]
Thanks!

> 1. How can we deal with relational data?

AceBase is not a relational database. You can store your data anywhere (eg in nested or isolated collections), no relations are enforced for any data.

> 2. How can we aggregate data?

You can query individual collections, index them to speed up those queries, and perform counts on them. If you need the average or sum of some field, run a query that returns the required fields and perform the aggregate function yourself in code. If you need aggregates for related data of another result set, simply perform another query with a filter on the relation and calculate their aggregate values.

> 3. Is it possible to use Acebase server with Bun https://bun.sh/

Bun looks really cool, I haven't tried it yet but it looks promising. If you would like to give it a try and let me know how it works, that would be awesome!

Let me know if you have other questions!
